File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.py
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        n = len(arr)

        arr.sort()

        t = 1

        for v in arr:
            if v >= t:
                t+=1

        return t-1
'''
        arr[0] = 1
        
        for i in range (1, n):
            if(arr[i] > arr[i-1] +1):
                arr[i] = arr[i-1] +1
        
        return arr[-1]
'''

File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.rs
impl Solution {
    pub fn maximum_element_after_decrementing_and_rearranging(arr: Vec<i32>) -> i32 {
        let mut sorted = arr.clone();

        sorted.sort();

        sorted[0] = 1;

        for i in 1..sorted.len() {
            if(sorted[i] > sorted[i-1] +1){
                sorted[i] = sorted[i-1] +1;
            }
        }

        return sorted[sorted.len() -1];
    }
}